File: Python/Endpoint Examples/Multipart Payload/batch-delete.py
from requests_toolbelt import MultipartEncoder
import requests
import json

delete_endpoint_url = 'https://api.pdfrest.com/delete'

mp_encoder_delete = MultipartEncoder(
    fields={
        'ids' : 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx, xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx',
    }
)

# Let's set the headers that the delete endpoint expects.
# Since MultipartEncoder is used, the 'Content-Type' header gets set to 'multipart/form-data' via the content_type attribute below.
headers = {
    'Accept': 'application/json',
    'Content-Type': mp_encoder_delete.content_type,
    'Api-Key': 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx' # place your api key here
}

print("Sending POST request to delete endpoint...")
response = requests.post(delete_endpoint_url, data=mp_encoder_delete, headers=headers)

print("Response status code: " + str(response.status_code))

if response.ok:
    response_json = response.json()
    print(json.dumps(response_json, indent = 2))
else:
    print(response.text)

# If you would like to download the file instead of getting the JSON response, please see the 'get-resource-id-endpoint.py' sample.

File: DotNET/Endpoint Examples/Multipart Payload/batch-delete.cs
var client = new HttpClient();
var request = new HttpRequestMessage(HttpMethod.Post, "https://api.pdfrest.com/delete");
request.Headers.Add("api-key", "xxxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx");
var content = new MultipartFormDataContent();
content.Add(new StringContent("xxxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx,xxxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"), "ids");
request.Content = content;
var response = await client.SendAsync(request);
response.EnsureSuccessStatusCode();
Console.WriteLine(await response.Content.ReadAsStringAsync());
